### PR TITLE
Added MS SQL Server to Legacy Store IT.

### LINKED ISSUE #16036: Add MSSQL to Legacy Store IT

### PR DESCRIPTION
**Pull request by miquelsi**
Closes #16036

### CODE REVIEW

**Comment by miquelsi** [comment]
There are failures in ConcurrencyTest for MS SQL Server. They will require a separate issue.

**Review by pedroigor — COMMENTED** [review]
@miquelsi The changes look good. Not sure though about what might be causing the failure on that test.

**Comment by miquelsi** [comment]
Logged failure in #16155

File: testsuite/integration-arquillian/pom.xml [hunk 1]
```diff
@@ -565,7 +565,7 @@
                 <keycloak.connectionsJpa.database>keycloak</keycloak.connectionsJpa.database>
                 <keycloak.connectionsJpa.user>sa</keycloak.connectionsJpa.user>
                 <keycloak.connectionsJpa.password>vEry5tron9Pwd</keycloak.connectionsJpa.password>
-                <keycloak.connectionsJpa.url>jdbc:sqlserver://${auth.server.db.host}:${docker.database.port};databaseName=${keycloak.connectionsJpa.database}</keycloak.connectionsJpa.url>
+                <keycloak.connectionsJpa.url>jdbc:sqlserver://${auth.server.db.host}:${docker.database.port};databaseName=${keycloak.connectionsJpa.database};integratedSecurity=false;encrypt=false;trustServerCertificate=true;</keycloak.connectionsJpa.url>
```
Comment (stianst) [inline testsuite/integration-arquillian/pom.xml:568]:
```suggestion
                <keycloak.connectionsJpa.url>jdbc:sqlserver://${auth.server.db.host}:${docker.database.port};databaseName=${keycloak.connectionsJpa.database};integratedSecurity=false;encrypt=false;trustServerCertificate=true;sendStringParametersAsUnicode=false;</keycloak.connectionsJpa.url>
```

**Review by pedroigor — APPROVED** [review]

**Pull request merged**